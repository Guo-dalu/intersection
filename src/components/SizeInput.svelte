<script lang="ts">
	import { Label, NumberInput, Helper } from 'flowbite-svelte'
	import { MIN_SIZE, MAX_SIZE } from '../constants'
	// for user experience, better not input a really large number
	// validate when submitting
	const TIPS = 'please input an interger between 1 to 10000000'
	export let size = 20
	export let label = ''
	export let style = ''
	$: isValid = Number.isInteger(size) && size >= MIN_SIZE && size <= MAX_SIZE
</script>

<div class={style + ' md:w-1/2 lg:w-1/3 font-light'}>
	<Label for={label} class="block mb-2">{label}</Label>
	<NumberInput
		aria-label={label}
		id={label}
		type="number"
		bind:value={size}
		min={MIN_SIZE}
		max={MAX_SIZE}
		color={isValid ? 'green' : 'red'}
		placeholder={TIPS}
	/>
	{#if !isValid}
		<Helper class="mt-2" color="red">{TIPS}</Helper>
	{/if}
</div>
