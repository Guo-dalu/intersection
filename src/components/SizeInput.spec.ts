import { expect, describe, it } from 'vitest'
import { render } from '@testing-library/svelte'
import SizeInput from './SizeInput.svelte'

describe('SizeInput Component', () => {
	it('should render correctly', () => {
		const { container } = render(SizeInput, { size: 100, label: 'size input' })
		expect(container).toMatchSnapshot()
	})
})
