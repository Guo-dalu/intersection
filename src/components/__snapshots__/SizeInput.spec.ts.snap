// Vitest Snapshot v1

exports[`SizeInput Component > should render correctly 1`] = `
<body>
  <div>
    <div
      class="md:w-1/2 lg:w-1/3 font-light"
    >
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300 block mb-2"
        for="size input"
      >
        size input
      </label>
      
      <!--&lt;Label&gt;-->
       
       
      <input
        class="block w-full disabled:cursor-not-allowed disabled:opacity-50 p-2.5 focus:ring-green-500 focus:border-green-500 dark:focus:border-green-500 dark:focus:ring-green-500 bg-green-50 text-green-900 placeholder-green-700 dark:text-green-400 dark:placeholder-green-500 dark:bg-gray-700 border-green-500 dark:border-green-400 text-sm rounded-lg"
        id="size input"
        max="1000000000"
        min="1"
        placeholder="better input an interger between 1 to 10000000"
        type="number"
      />
       
      
      
      <!--&lt;Wrapper&gt;-->
      <!--&lt;Input&gt;-->
      <!--&lt;NumberInput&gt;-->
       
    </div>
    <!--&lt;SizeInput&gt;-->
  </div>
</body>
`;
